### ISSUE TITLE
Version 0.28 plans

### ISSUE THREAD

**Issue by nihohit** [redis-rs/redis-rs#1409]
### Bug fixes
- [x] Sync cluster - reconnect after complete disconnect - https://github.com/redis-rs/redis-rs/pull/1233
- [x] MpxConnection: Allow pipelines to continue after error - https://github.com/redis-rs/redis-rs/pull/1293
- [x] Do not retry requests that have been dropped by the user - https://github.com/redis-rs/redis-rs/pull/1318

### docs improvements
- [x] Improve docs - https://github.com/redis-rs/redis-rs/pull/1339
- [x] Add doc clarifying behavior of invoke_script - https://github.com/redis-rs/redis-rs/pull/1396
- [x] Improve the docs on connection pool usage - https://github.com/redis-rs/redis-rs/pull/1411
- [x] Clarify RESP3 subscription documentation https://github.com/redis-rs/redis-rs/pull/1436

### Pubsub improvements prerequisites
- [x] Choose sleep function according to active runtime - https://github.com/redis-rs/redis-rs/pull/1369
- [x] Use runtime agnostic retry mechanism - https://github.com/redis-rs/redis-rs/pull/1392
- [x] ConnectionManager: reconnect on disconnect pushes - https://github.com/redis-rs/redis-rs/pull/1407

### Pubsub improvements
- [x] Async push messages: Allow for more channel types - https://github.com/redis-rs/redis-rs/pull/1295
- [x] Add ping support to commands & pubsubs - https://github.com/redis-rs/redis-rs/pull/1341
- [ ] Add async pubsub manager - https://github.com/redis-rs/redis-rs/pull/1393
- [x] ConnectionManager: Support automatic resubscription - https://github.com/redis-rs/redis-rs/pull/1408
- [x] add pubsub support to async cluster - https://github.com/redis-rs/redis-rs/pull/1449

### Nice to have
- [x] Use cargo-nextest for testing - https://github.com/redis-rs/redis-rs/pull/1336
- [x] test in CI against valkey 8 - https://github.com/redis-rs/redis-rs/pull/1362
- [x] Move server creation logic to redis-test - https://github.com/redis-rs/redis-rs/pull/1363
- [x] Add keywords to increase visibility - https://github.com/redis-rs/redis-rs/pull/1425
- [ ] https://github.com/redis-rs/redis-rs/pull/1437

**Comment by altanozlu** [comment]
Hi what is the plan for `add pubsub support to async cluster - TBD` ? Also can you pin next version issues ?

**Comment by nihohit** [comment]
> what is the plan for add pubsub support to async cluster - TBD ?

I've already implemented it a couple of months ago, but I wanted to rebase it on the work here. I hope to rebase it this week and see if something got broken, and if the general design still makes sense. Hope fully I'll open a PR soon.

> can you pin next version issues ?

I haven't looked that forward yet. Do you have any suggestions, except for caching?

**Comment by altanozlu** [comment]
> I've already implemented it a couple of months ago, but I wanted to rebase it on the work here. I hope to rebase it this week and see if something got broken, and if the general design still makes sense. Hope fully I'll open a PR soon.
>
Perfect!

> > can you pin next version issues ?
> 
> I haven't looked that forward yet. Do you have any suggestions, except for caching?
>
What I meant by pinning is [this](https://docs.github.com/en/issues/tracking-your-work-with-issues/administering-issues/pinning-an-issue-to-your-repository), so next release version plan will be always at top.

**Comment by nihohit** [comment]
oh, ok. done :)

**Comment by elichai** [comment]
Will it be possible to create a `PubSub` struct from `ConnectionManager`?

**Comment by nihohit** [comment]
@elichai if your server's version is 6.2 or above, you can already use the RESP3 push sender -
https://docs.rs/redis/latest/redis/#resp3-async-pubsub
ATM it doesn't support automatic resubscription, but once 0.28 will be released (hopefully in the next couple of weeks) it will.

**Comment by elichai** [comment]
@nihohit Sorry, these are technically 2 separate questions:
1. About how to use the `ConnectionManager` to do pub/sub, if I understand correctly it's a combination of `.subscribe` calls and [AsyncPushSender](https://docs.rs/redis/latest/redis/aio/trait.AsyncPushSender.html).
2. About support for a dedicated `PubSub` sub-type to be created from the `ConnectionManager`.

But I think my confusion stemed from not understanding that each `PubSub` struct is its own connection and you cannot share the same connection with multiple separate subscriptions

**Comment by nihohit** [comment]
Yes. We might introduce a reconnecting PubSub in the future, but it will be a separate entity, not a converted `ConnectionManager`.
